scipy/_lib/_util.py

```diff
@@ -11,6 +11,53 @@
 import numpy as np
 
 
+def _valarray(shape, value=np.nan, typecode=None):
+    """Return an array of all value.
+    """
+
+    out = np.ones(shape, dtype=bool) * value
+    if typecode is not None:
+        out = out.astype(typecode)
+    if not isinstance(out, np.ndarray):
+        out = np.asarray(out)
+    return out
```

Comment:
I don't know the story behind `valarray`, but your suggested replacement treats `value` less flexibly. Currently valarray allows any `value` that can be broadcasted onto an array with the provided shape, whereas `.fill(value)` would require a scalar `value`.